dali/operators/decoder/video/video_decoder_base.h

```diff
@@ -15,121 +15,437 @@
+    int stride = GetStride(sample_idx);
+    int available_frames = std::max(0, total_num_frames - start_frame);
+    int available_strided_frames = (available_frames + stride - 1) / stride;
+    return available_strided_frames;
+  }
+
+  /**
+   * @brief Returns the number of frames to be decoded for a given sample
    *
-   * @param output The SampleView in which the decoded sequence will be put.
-   * @param sample_idx Index of the encoded video that shall be decoded.
-   * @param num_frames How many frames shall be decoded.
-   * @param pad_value What to (optionally) pad the partial sequence with.
-   * @return False, if less than `num_frames` have been decoded.
+   * The sequence length is determined by the following priority:
+   * 1. If `frames` argument exists, use its length
+   * 2. If `sequence_length` argument exists, use its value, capped by available frames if boundary
+   * type is ISOLATED, and taking into account the start frame and stride,
+   * 3. Otherwise, use the total number of available frames (taking into account the start frame and
+   * stride)
    */
-  bool DecodeFrames(SampleView<OutBackend> output, int64_t sample_idx, int64_t num_frames,
-                    std::optional<SampleView<OutBackend>> pad_value = std::nullopt,
-                    std::optional<cudaStream_t> stream = std::nullopt) {
-    auto &frames_decoder = *frames_decoders_[sample_idx];
-    int64_t frame_size = frames_decoder.FrameSize();
+  int GetSequenceLength(int sample_idx) const {
+    int sequence_len;
+    if (frames_.HasValue()) {
+      sequence_len = frames_[sample_idx].shape[0];
+    } else if (sequence_length_.HasValue()) {
+      sequence_len = sequence_length_[sample_idx].data[0];
+      if (boundary_type_ == boundary::BoundaryType::ISOLATED) {
+        sequence_len = std::min(sequence_len, GetAvailableNumFrames(sample_idx));
+      }
+    } else {
+      sequence_len = GetAvailableNumFrames(sample_idx);
+    }
+    return sequence_len;
+  }
 
-    DALI_ENFORCE(!pad_value.has_value() || pad_value->shape().num_elements() == frame_size,
-                 make_string("Provided pad_value has improper number of elements. Expected: ",
-                             frame_size, "; Actual: ", pad_value->shape().num_elements()));
+  bool SetupImpl(std::vector<OutputDesc> &output_desc, const Workspace &ws) override {
+    const auto &input = ws.Input<InBackend>(0);
 
-    uint8_t *output_data = output.template mutable_data<uint8_t>();
+    // Validate input
+    DALI_ENFORCE(input.type() == DALI_UINT8, "Type of the input buffer must be uint8.");
+    DALI_ENFORCE(input.sample_dim() == 1, "Input buffer must be 1-dimensional.");
+    for (int64_t i = 0; i < input.num_samples(); ++i) {
+      DALI_ENFORCE(input[i].shape().num_elements() > 0,
+                   make_string("Incorrect sample at position: ", i, ". ",
+                               "Video decoder does not support empty input samples."));
```

Comment:
DO you know if image decoder has a similar check?

Reply:
no such check but it'll eventually fail due to parsing an incorrect stream